### ISSUE TITLE
No execution time accumulated

### ISSUE THREAD

**Issue by LucasAdams** [Theano/Theano#3409]
Default linker / vm does not allow profiling information to be gathered.

Setting `profile=True` in .theanorc is not sufficient and printing the profile information results in the message "No execution time accumulated (hint: try config profiling.time_thunks=1)". This is output by [ProfileStats.summary](https://github.com/Theano/Theano/blob/master/theano/compile/profiling.py#L1287). The hint of setting `[profiling]\n time_thunks=1` doesn't fix the issue.

Debugging:
ProfileStats.summary requires the apply_time dictionary, a mapping of nodes to floats of running time. This dictionary is only modified in [VM.update_profile](https://github.com/Theano/Theano/blob/master/theano/gof/vm.py#L228) which is executed only by  [Function.__call__](https://github.com/Theano/Theano/blob/master/theano/compile/function_module.py#L917) on the Function.fn variable after the function has finished executing. However Function.fn by default is an instance of [streamline_default_f](https://github.com/Theano/Theano/blob/master/theano/gof/link.py#L644) which doesn't have this function (Function.__call__ checks to ensure no error is generated).

The solution is to use the [newer](https://github.com/Theano/Theano/blob/master/theano/gof/vm.py#L5) vm rather than linker interface. This can be enabled by setting the `mode` [configuration option](https://github.com/Theano/Theano/blob/master/theano/compile/mode.py#L69) to `vm` or `vm_nogc`which enables the LoopGC or Loop VMs respectively. The C-based VMs may also be able to profile (`cvm` or `cvm_nogc`) but I haven't experimented, the comment indicates [no](https://github.com/Theano/Theano/blob/master/theano/gof/lazylinker_c.c#L23) but the code indicates maybe [1](https://github.com/Theano/Theano/blob/master/theano/gof/lazylinker_c.c#L496) [2](https://github.com/Theano/Theano/blob/master/theano/gof/lazylinker_c.c#L526).

In summary, for profiling:

```
linker=vm
profile=True
[profiling]
time_thunks=1
```

I am unsure how this should best be addressed. Either the default VM needs to be `vm` or the documentation should be updated. It is unclear to me why we have the apparently legacy linker system.

As an aside, to print profiling information without closing (i)python, use `f.profile.summary()`.

**Comment by nouiz** [comment]
Closing. Now we have a new Theano flag that allow to print only global stats at the end of the process.

Also, now the defeault linker (cvm) allow profiing.
